SilverSolver/assignment3/word_search.py

```diff
@@ -0,0 +1,49 @@
+from bisect import bisect_left
+from dictionary import Dictionary
+from typing import List, Tuple
+
+def empty_matrix(m: int, n: int, value: bool) -> List[List[bool]]:
+    # value should has immutable type!
+    result = []
+    for i in range(m):
+        result.append([value] * n)
+    return result
+
+def word_search(grid: List[List[str]], dictionary: Dictionary) -> List[str]: 
+    # -- Should we check grid correctness?
+    def neighbours(coords: Tuple[int]) -> Tuple[Tuple[int]]:
+        a, b = coords
+        neigh_coords = [(a + i, b + j) for i in [-1, 1, 0] for j in [-1, 1, 0]]
+        del neigh_coords[-1]
+        return filter(lambda x: x[0] >= 0 and x[1] >= 0 and x[0] < m and x[1] < n, neigh_coords)
+    
+    def search(coords: Tuple[int]):
```

Comment:
Unlike List[type] that takes one type and the elements should be matching the type, Tuple takes as many types as there are elements. Tuples are not really supposed to be of variable length, especially since they are immutable.

I think the correct type is Tuple[int, int]